Sources/WebSocketKit/WebSocket.swift

```diff
@@ -63,6 +75,32 @@ public final class WebSocket {
         self.onPingCallback = callback
     }
 
+    public func onEvent(_ identifier: String, _ handler: @escaping (WebSocket) -> Void) {
+        events[identifier] = { ws, data in
+            handler(ws)
+        }
+    }
+
+    public func onEvent<T>(_ identifier: String, _ type: T.Type, _ handler: @escaping (WebSocket, T) -> Void) where T: Codable {
+        onEvent(identifier, handler)
+    }
+
+    public func onEvent<T>(_ identifier: String, _ handler: @escaping (WebSocket, T) -> Void) where T: Codable {
+        events[identifier] = { [weak self] ws, data in
+            guard let self = self else { return }
+            do {
+                let res = try JSONDecoder().decode(WebSocketEvent<T>.self, from: data)
```

Comment:
We should use the decoder from above
```suggestion
                let res = try self.decoder.decode(WebSocketEvent<T>.self, from: data)
```